src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -1,27 +1,24 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
-using System.Threading.Tasks;
-using System.Diagnostics.Contracts;
+using System.Diagnostics;
 
 namespace System.Net.Http
 {
     public class HttpMethod : IEquatable<HttpMethod>
     {
-        private string _method;
-
-        private static readonly HttpMethod s_getMethod = new HttpMethod("GET");
-        private static readonly HttpMethod s_putMethod = new HttpMethod("PUT");
-        private static readonly HttpMethod s_postMethod = new HttpMethod("POST");
-        private static readonly HttpMethod s_deleteMethod = new HttpMethod("DELETE");
-        private static readonly HttpMethod s_headMethod = new HttpMethod("HEAD");
-        private static readonly HttpMethod s_optionsMethod = new HttpMethod("OPTIONS");
-        private static readonly HttpMethod s_traceMethod = new HttpMethod("TRACE");
-
-        // Don't expose CONNECT as static property, since it's used by the transport to connect to a proxy. 
+        private readonly string _method;
+        private int _hashcode;
```

Comment:
Also, regarding this:

>My only pause is that some developers may be unaware of the static properties, and when they see a method that takes an  HttpMethod  parameter, they may immediately try to "new one up" and just use the constructor without knowing about the static properties.

That almost never happens.  We usually never see developers doing `new HttpMethod()` especially with the standard method verbs.